packages/core/src/types.ts

```diff
@@ -545,6 +545,7 @@ export type OutgoingMessage = {
 type Filter<T, U> = T extends U ? T : never;
 type SelectedValueTypes = Filter<ComponentPropertyValue['type'], 'UnboundValue' | 'BoundValue'>;
 
+export type RequestReadOnlyModePayload = undefined;
```

Comment:
Considering the idea of passing a tree _as_ the payload when requesting read-only mode.  I don't see why this should need to be a two step process of requesting read only mode, then requesting the component tree to render, but interested to hear what others think.

For example, in user interface, when opening up the version history modal, send the post message:
```js
const { tree } = useTreeStore();
const clonedTree = cloneDeep(tree);
const backwardsCompatibleClonedTree = createBackwardCompatibleTree(clonedTree);

sendMessage(CANVAS_ACTIONS.RequestReadOnlyMode, {
  tree: backwardsCompatibleClonedTree
  locale
});  
```

Unless anyone has any objections I'm going to implement that pattern now.

Reply:
DONE!